src/async_impl/h3_client/pool.rs

```diff
@@ -221,11 +221,17 @@ impl PoolClient {
 
         let mut stream = self.inner.send_request(req).await?;
 
-        match req_body.as_bytes() {
-            Some(b) if !b.is_empty() => {
-                stream.send_data(Bytes::copy_from_slice(b)).await?;
+        let mut req_body = Pin::new(&mut req_body);
+        loop {
```

Comment:
This should probably be done on a separate task, so that sending the full request body doesn't block getting the response, the streams can be fully duplex.